### PR TITLE
create a map of FeatureSets

### PR DESCRIPTION
**Pull request by rphillips**
From initial work on implementing FeatureGates, an initial pass would require each module to switch on the enabled FeatureSet to fetch the Enabled and Disabled slices. We can simplify this a bit by using a map to fetch the Enabled/Disabled strings.

/cc @sjenning @derekwaynecarr @deads2k

### CODE REVIEW

File: config/v1/types_features.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ type Features struct {
 type FeatureSet string
 
 var (
+	// Default feature set that allows upgrades.
+	Default FeatureSet = "Default"
```
Comment (deads2k) [inline config/v1/types_features.go]: no. this is just empty string.

Comment (rphillips) [reply config/v1/types_features.go]: @deads2k fixed in 57a1337

File: config/v1/types_features.go [hunk 2]
```diff
@@ -44,22 +47,22 @@ type FeaturesList struct {
...
+}
 
-	// DefaultDisabledKubeFeatures is a list of features to turn off when the default featureset is turned on.  Only the exceptions
-	// are listed here.  The feature gates left in their default state not listed.
-	DefaultDisabledKubeFeatures = []string{
-		"PersistentLocalVolumes", // disable local volumes for 4.0, owned by sig-storage/hekumar@redhat.com
-	}
-)
+// FeatureSets Contains a map of Feature names to Enabled/Disabled Features.
+// NOTE: The caller needs to make sure to check for the existence of the value
```
Comment (deads2k) [inline config/v1/types_features.go]: Explain the case where it won't exist.  Upgrades with new enumation values will be a common one.

Comment (rphillips) [reply config/v1/types_features.go]: Added in 645f70d

**Pull request merged**